build-data-capturing-gradle-samples/capture-processor-arch/gradle-processor-arch.gradle

```diff
@@ -0,0 +1,66 @@
+import java.nio.charset.Charset
+import java.util.concurrent.TimeUnit
+
+/**
+ * This Gradle script captures the processor architecture
+ * and adds these as a custom value.
+ * This should be applied to the root project:
+ * <code> apply from: file('gradle-processor-arch.gradle') </code>
+ */
+
+def buildScanApi = project.extensions.findByName('buildScan')
+if (!buildScanApi) {
+  return
+}
+
+buildScan.background { api ->
+  captureProcessorArch(api)
+}
+
+static void captureProcessorArch(def api) {
+  def osName = System.getProperty("os.name")
+  api.value("os.name", osName)
+
+  def osArch = System.getProperty("os.arch")
+  api.value("os.arch", osArch)
+
+  if (isDarwin(osName)) {
+    if (isM1()) {
+      api.tag("M1")
+    }
+    if (isTranslatedByRosetta()) {
```

Comment:
Is this an `if` on purpose or should it be a `else if`?

Reply:
It's an `if` on purpose as I'm unsure of the behavior of `uname -p` when Rosetta comes into play.

Reply:
Please decide with @runningcode. I'd like to stay out of the details.

Reply:
I think the `if else` would clarify the intention here if logically only one tag could be applied.

Reply:
just done

Reply:
Just double checking here Jerome: The link you sent describes the behavior of `uname -m` not `uname -p`. Can you ask someone on the team to test this with and without translation enabled?

Reply:
`uname -p` would just hides a few details but lead to the same conclusion
https://stackoverflow.com/questions/65259300/detect-apple-silicon-from-command-line

Script has already been tested (not translated yet)